libexec/pyenv

```diff
@@ -72,11 +73,10 @@ fi
 
 if [ -z "${PYENV_DIR}" ]; then
   PYENV_DIR="$PWD"
-else
-  cd "$PYENV_DIR" 2>/dev/null || abort "cannot change working directory to \`$PYENV_DIR'"
-  PYENV_DIR="$PWD"
-  cd "$OLDPWD"
 fi
+
+PYENV_DIR=$(cd "$PYENV_DIR" && echo "$PWD") || abort "cannot change working directory to \`$PYENV_DIR'"
```

Comment:
Why is this necessary after all?
Could also check only if it is a directory then?!

Reply:
No idea, I'm trying to respect the current code as closely as possible

Reply:
I've added that code back in to test `-d` and `-x`.  I'm not sure why the check is there in the first place, but there is a test for this specific check.

Reply:
> to get the absolute path

This is what I've meant mostly.
At least there does not seem to be a failing test when removing the `PYENV_DIR=$(cd "$PYENV_DIR" && echo "$PWD")`.
Since this spawns a subshell, it is not really cheap, and should be avoided if not necessary.